topdown/crypto.go

```diff
@@ -425,6 +558,41 @@ func verifyX509CertificateChain(certs []*x509.Certificate) ([]*x509.Certificate,
 	return chains[0], nil
 }
 
+func verifyX509CertificateChainWithOpts(certs []*x509.Certificate, vo verifyOptions) ([]*x509.Certificate, error) {
```

Comment:
I think that rather than making a new function here we can extend verifyX509CertificateChain to support the passing of an existing options struct that is extended if present or initialised fresh if not.

Reply:
👍 sounds like a good plan, thanks.